source/connectors/kafka-source.txt

```diff
@@ -0,0 +1,265 @@
+
+  - For insert and replace operations, it contains the new document being
+    inserted or replacing the existing document.
+  - For update operations, it contains the complete document that is being
+    updated at some point in time after the update occurred. If the
+    document was deleted since the update, it contains a null value.
+
+* The ``documentKey`` contains either the ``_id`` field of the document
+  that was updated or all the components of a shard key for sharded
+  collections.
+
+* The ``txnNumber`` and ``lsid`` identify the transaction if the change
+  occurred within one.
+
+.. code-block:: json
+
+   {
+     _id: { <BSON Object> },
+     "operationType": "<operation>",
+     "fullDocument": { <document> },
+     "ns": {
+       "db": <database>,
+       "coll": <collection>
+     },
+     "to": {
+       "db": <database>,
+       "coll": <collection>
+     },
+     "documentKey": {
+       _id: <value>
+     },
+     "updateDescription": {
+       "updatedFields": { <document> },
+       "removedFields": [ <field>, ... ]
+     },
+     "clusterTime": <Timestamp>,
+     "txnNumber": <NumberLong>,
+     "lsid": {
+       "id": <UUID>,
+       "uid": <BinData>
+     }
+   }
+
+Source Connector Configuration Properties
+-----------------------------------------
+
+The MongoDB Kafka Source Connector uses the following settings to create
+change streams and customize the output to save to the Kafka cluster. For
+an example source connector configuration file, see
+`MongoSourceConnector.properties
+<https://github.com/mongodb/mongo-kafka/blob/master/config/MongoSourceConnector.properties>`_.
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Name
+     - Type
+     - Description
+
+   * - connection.uri
+     - string
+     - | A :manual:`MongoDB connection URI string </reference/connection-string/#standard-connection-string-format>`.
+       | **Default**: ``mongodb://localhost:27017,localhost:27018,localhost:27019``
+       | **Accepted Values**: A valid MongoDB connection URI string
+
+   * - database
+     - string
+     - | Name of the database to watch for changes. If not set, all databases are watched.
+       | **Default**: ""
+       | **Accepted Values**: A single database name
+
+   * - collection
+     - string
+     - | Name of the collection in the database to watch for changes.
+       | The collection in the database to watch. If not set then all collections will be watched.
+       | **Default**: ""
+       | **Accepted Values**: A single collection name
+
+   * - publish.full.document.only
+     - boolean
+     - | Only publish the changed document instead of the full change stream document. Sets the ``change.stream.full.document=updateLookup`` automatically so updated documents will be included.
+
+       .. seealso:: :ref:`Publish full document example <full-document-example>`
+
+       | **Default**: false
+       | **Accepted Values**: ``true`` or ``false``
+
+   * - pipeline
+     - string
```

Comment:
shouldn't this be `[]string` or `[string]`?

Reply:
Since these are config files, the convention I adopted was to use "string" for JSON document arrays since there are no explicit data structures. The "Accepted Values" field describes the contents of the field in more detail.

There was a comment a while back in which I was asking a reviewer whether including that field was necessary; based on the current state, the result of that conversation must have been that it was still valid to include.

E.g. 
https://github.com/mongodb/docs-ecosystem/pull/549/files#diff-774db6e64574a30a2870b1d1b33a9638R111
https://github.com/mongodb/docs-ecosystem/pull/549/files#diff-774db6e64574a30a2870b1d1b33a9638R123
...

This is also the convention that was used in the original Kafka Connector docs.
Does that make sense to use, or should we adopt a new convention for config files?

Reply:
Ok, makes sense. Just wanted to make sure there was intention.